crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1543,7 +1544,10 @@ where
                                         tx_digest = ?tx_digest,
                                         "Got quorum for validators handle_certificate."
                                     );
-                                    return Ok(ReduceOutput::End(state));
```

Comment:
hmmm this was changed to ReduceOutput::End on purpose, otherwise the latency ends up being equal to the slowest: https://github.com/MystenLabs/sui/pull/4606
I thought the fix to this kind of problems is to move it to separate tokio thread (as the comment says in the PR), wonder why it didn't work

Reply:
> wonder why it didn't work

What's the "it" that you're referring to? We could move the work into a new task, but that hasn't been tried yet in this case.

The problem here is not that we sent a request to the validator and then the connection was dropped before it could finish. The problem is that we got an error response from the validator, we need to do client-side work to bring the validator up to date, but we return from the function before doing that work.

Reply:
I probably just misunderstood, but I was referring to this comment:
```
 also moved sync_certificate_to_authority to run in a tokio task so that we don't abort the syncing process mid way once we have enough votes.
```

Reply:
FYI, the above comment is referring to this change in authrority_aggregator.rs where authority sync is now happening in a tokio task:
```
let authority_clients = self.authority_clients.clone();
if let Ok(res) = timeout(total_timeout, tokio::spawn(async move {
    Self::sync_certificate_to_authority_with_timeout(
       &committee,
```